cmd/ttn-lw-cli/commands/end_devices.go

```diff
@@ -349,13 +350,22 @@ var (
 			}
 			paths := util.UpdateFieldMask(cmd.Flags(), setEndDeviceFlags, attributesFlags())
 
+			abp, _ := cmd.Flags().GetBool("abp")
+			multicast, _ := cmd.Flags().GetBool("multicast")
 			var device ttnpb.EndDevice
 			if inputDecoder != nil {
 				decodedPaths, err := inputDecoder.Decode(&device)
 				if err != nil {
 					return err
 				}
 				paths = append(paths, ttnpb.FlattenPaths(decodedPaths, endDeviceFlattenPaths)...)
+
+				if ttnpb.ContainsField("supports_join", paths) {
+					if abp && device.SupportsJoin {
```

Comment:
```suggestion
					if (abp || multicast) && device.SupportsJoin {
```

Reply:
Multicast implies ABP, that's why I originally suggested setting `abp = abp || multicast` above